File: Backend-RS/models/perfil_interes.py
from sqlmodel import Field, SQLModel, Relationship
from typing import Optional
from sqlalchemy import UniqueConstraint
from config.database import get_session

class PerfilInteresBase(SQLModel):
    IdPerfil: int
    IdInteres: int
class PerfilInteres(SQLModel, table=True):
    __table_args__ = (UniqueConstraint("IdPerfil", "IdInteres", name="uq_perfil_interes"),)

    IdPerfilInteres: Optional[int] = Field(default=None, primary_key=True)
    IdPerfil: int = Field(foreign_key="perfil.IdPerfil")
    IdInteres: int = Field(foreign_key="intereses.IdInteres")

    perfil: "Perfil" = Relationship(back_populates="intereses_rel")
    interes: "Intereses" = Relationship(back_populates="perfiles_rel")
class PerfilInteresCreate(PerfilInteresBase):
    pass

class PerfilInteresPublic(PerfilInteresBase):
    IdPerfilInteres: int

class PerfilInteresUpdate(PerfilInteresBase):
    pass

PerfilInteresPublic.model_rebuild()

File: frontendrs/lib/data/models/perfil_interes.dart
class PerfilInteresBase {
  final int idPerfil;
  final int idInteres;

  PerfilInteresBase({required this.idPerfil, required this.idInteres});

  factory PerfilInteresBase.fromJson(Map<String, dynamic> json) {
    return PerfilInteresBase(
      idPerfil: json['IdPerfil'],
      idInteres: json['IdInteres'],
    );
  }

  Map<String, dynamic> toJson() {
    return {'IdPerfil': idPerfil, 'IdInteres': idInteres};
  }
}

class PerfilInteres extends PerfilInteresBase {
  final int? idPerfilInteres;

  PerfilInteres({
    this.idPerfilInteres,
    required int idPerfil,
    required int idInteres,
  }) : super(idPerfil: idPerfil, idInteres: idInteres);

  factory PerfilInteres.fromJson(Map<String, dynamic> json) {
    return PerfilInteres(
      idPerfilInteres: json['IdPerfilInteres'],
      idPerfil: json['IdPerfil'],
      idInteres: json['IdInteres'],
    );
  }

  @override
  Map<String, dynamic> toJson() {
    return {
      'IdPerfilInteres': idPerfilInteres,
      'IdPerfil': idPerfil,
      'IdInteres': idInteres,
    };
  }
}

class PerfilInteresCreate {
  final int idPerfil;
  final int idInteres;

  PerfilInteresCreate({required this.idPerfil, required this.idInteres});

  Map<String, dynamic> toJson() {
    return {'IdPerfil': idPerfil, 'IdInteres': idInteres};
  }
}

class PerfilInteresPublic extends PerfilInteresBase {
  final int idPerfilInteres;

  PerfilInteresPublic({
    required this.idPerfilInteres,
    required int idPerfil,
    required int idInteres,
  }) : super(idPerfil: idPerfil, idInteres: idInteres);

  factory PerfilInteresPublic.fromJson(Map<String, dynamic> json) {
    return PerfilInteresPublic(
      idPerfilInteres: json['IdPerfilInteres'],
      idPerfil: json['IdPerfil'],
      idInteres: json['IdInteres'],
    );
  }

  @override
  Map<String, dynamic> toJson() {
    return {
      'IdPerfilInteres': idPerfilInteres,
      'IdPerfil': idPerfil,
      'IdInteres': idInteres,
    };
  }
}

class PerfilInteresUpdate {
  final int idPerfil;
  final int idInteres;

  PerfilInteresUpdate({required this.idPerfil, required this.idInteres});

  Map<String, dynamic> toJson() {
    return {'IdPerfil': idPerfil, 'IdInteres': idInteres};
  }
}
